zebra/rt_netlink.c

```diff
@@ -1805,6 +1817,7 @@ netlink_route_multipath (int cmd, struct prefix *p, struct rib *rib,
                 SET_FLAG (nexthop->flags, NEXTHOP_FLAG_FIB);
             }
         }
+
       if (src)
         addattr_l (&req.n, sizeof req, RTA_PREFSRC, &src->ipv4, bytelen);
```

Comment:
This code should work only for multipath case.
I don't have anything related ipv6 multipath in the codebase yet.
But I could add this into the code. It doesn't matter, honestly.

---
In reply to: [86168202](https://github.com/Azure/sonic-quagga/pull/4#discussion_r86168202) [](ancestors = 86168202)